src/query_frontend/src/plan.rs

```diff
@@ -125,15 +138,15 @@ impl QueryPlan {
     /// Note: When it timestamp filter evals to false(such as ts < 10 and ts >
     /// 100), it will return None, which means no valid time range for this
     /// query.
-    fn extract_time_range(&self) -> Option<TimeRange> {
-        let ts_column = if let Some(v) = self.find_timestamp_column() {
+    fn extract_time_range(&self) -> Result<Option<TimeRange>> {
+        let ts_column = if let Some(v) = self.find_timestamp_column()? {
             v
         } else {
             warn!(
                 "Couldn't find time column, plan:{:?}, table_name:{:?}",
                 self.df_plan, self.table_name
             );
-            return Some(TimeRange::min_to_max());
+            return Ok(Some(TimeRange::min_to_max()));
```

Comment:
Consider return `Err(...)` rather than `OK(...)`, expose err to upper layers.
```suggestion
            return Ok(Some(TimeRange::min_to_max()));
```

Reply:
The error is already return in `self.find_timestamp_column()?`, the else branch is when query range is not found, such as `select 1`